File: Selenium/Project/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    # Navigate to the webpage
    driver.get("http://alchemy.hguy.co/orangehrm")

    # Find and fill in the username and password fields
    username = driver.find_element(By.XPATH, "//input[@id = 'txtUsername']")
    password = driver.find_element(By.XPATH, "//input[@id = 'txtPassword']")
    username.send_keys("orange")
    password.send_keys("orangepassword123")

    # Click the login button
    driver.find_element(By.XPATH, "//input[@id = 'btnLogin']").click()
    # Find and click on "My Info"
    myInfo = driver.find_element(By.CSS_SELECTOR, "a#menu_pim_viewMyDetails")
    myInfo.click()

    # Find and click on "Qualifications"
    qualification = driver.find_element(By.XPATH, "//*[@id='sidenav']/li/a[text()='Qualifications']")
    qualification.click()

    # Find and click on "Add Work Experience"
    add = driver.find_element(By.XPATH, "//*[@id='addWorkExperience']")
    add.click()

    # Fill in company and job title
    company = driver.find_element(By.XPATH, "//*[@id='experience_employer']")
    company.send_keys("IBM India")

    jobTitle = driver.find_element(By.XPATH, "//*[@id='experience_jobtitle']")
    jobTitle.send_keys("Test Specialist")

    # Click the save button
    driver.find_element(By.XPATH, "//*[@id='btnWorkExpSave']").click()
    driver.quit()

File: Selenium/Project/Activity7.java
package Projects_HRM;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity7 {
    static WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public static void setup() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        //Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        driver.get("http://alchemy.hguy.co/orangehrm");
    }

    @Test(priority = 1)
    public void login() throws InterruptedException {
        WebElement userName = driver.findElement(By.xpath("//*[@id='txtUsername']"));
        WebElement password = driver.findElement(By.xpath("//*[@id='txtPassword']"));
        WebElement loginBtn = driver.findElement(By.xpath("//*[@id='btnLogin']"));
        //--------- Login
        userName.sendKeys("orange");
        password.sendKeys("orangepassword123");
        loginBtn.click();
    }

    @Test(priority = 2)
    public void clickMyInfo() {

        WebElement myInfo = driver.findElement(By.cssSelector("a#menu_pim_viewMyDetails"));
        myInfo.click();

    }

    @Test(priority = 3)
    public void clickQualification() {
        WebElement myInfo = driver.findElement(By.cssSelector("a#menu_pim_viewMyDetails"));
        myInfo.click();
        WebElement qualification = driver.findElement(By.xpath("//*[@id = 'sidenav']/li/a[text()='Qualifications']"));
        qualification.click();
        WebElement add = driver.findElement(By.xpath("//*[@id = 'addWorkExperience']"));
        add.click();
        WebElement company = driver.findElement(By.xpath("//*[@id = 'experience_employer']"));
        company.sendKeys("IBM India");
        WebElement jobTitle = driver.findElement(By.xpath("//*[@id = 'experience_jobtitle']"));
        jobTitle.sendKeys("Test Specialist");
        driver.findElement(By.xpath("//*[@id = 'btnWorkExpSave']")).click();
    }


    @AfterClass
    public static void afterMethod() {
        //Close the browser
        driver.quit();
    }
}
